### ISSUE TITLE
`sort: flip` option for structure fields

### ISSUE THREAD

**Issue by lukasbestle** [getkirby/panel#722]
See [this topic in the forum](https://forum.getkirby.com/t/add-new-builder-structure-entries-to-the-top-not-bottom-of-entries/2958?u=lukasbestle).

Could be implemented with a `sort: flip` option that would change the display order (and still save them the old way; the template code can use `->flip()` as well).

**Comment by texnixe** [comment]
And complement that with an option to auto-sort by a given field, e.g. `sort: eventTitle`.

**Comment by lukasbestle** [comment]
+1!

**Comment by louiswalch** [comment]
+1

**Comment by medienbaecker** [comment]
+1

**Comment by distantnative** [comment]
If we have `sort: eventTile`, should flip not be a separate options like `flip: true`? to choose a sort field and to flip at the same time.

**Comment by texnixe** [comment]
Yes, two separate options make perfect sense.

**Comment by distantnative** [comment]
I started a PR - still work in progress. Sorting and flipping in the panel works fine already. But now I'm actually wondering what should happen with the values in the file. Should they be sorted or just left as it (order of adding) and only sorted when applying e.g. `sortBy` in the template?

**Comment by lukasbestle** [comment]
I think they should be saved just like before. The sorting and flipping should only dynamically change the display order.

**Comment by distantnative** [comment]
Okay, then the PR should be ready.

**Comment by lukasbestle** [comment]
Well, I don't know. Maybe there are good arguments for saving the order directly. Let's wait what other people think.

**Comment by distantnative** [comment]
Yea. If there are voices for doing it differently, it would probably be another PR.
The problem with sorting the file is, that the way the structure field works would only allow to trigger a sort when it's modified (added, changed, deleted). So if you have a structure field with some entries and you change the blueprint, simply updating the page would not change the order in the file. Only if you actually change an entry of the structure field and then update the page. And that seems to me quite confusing.

**Comment by lukasbestle** [comment]
Yes, that's true. On the other hand, people may be confused if the order is different when accessing the items in the templates.

**Comment by texnixe** [comment]
I'd keep the order in the files as they are. The controller should take care of how the entries are displayed on the website and the editor will not usually see the content of the text files directly.

**Comment by distantnative** [comment]
:+1: - was also thinking about a good syntax to have multiple fields for sorting (as suggested in #652).

```
sort: 
  - ordernumber
  - secondfield
```

But how would we specify ASC/DESC in such a YAML syntax?

```
sort: 
  - ordernumber, asc
  - secondfield, desc
```

**Comment by texnixe** [comment]
Unfortunately, the fix never made it into core, still waiting to have this (had to apply the fix manually in the core for some projects).

**Comment by lukasbestle** [comment]
I have added the PR to the 2.5 milestone, we will at least take a look at it for that release.

**Comment by lukasbestle** [comment]
This feature is now available in v2.5.0.
